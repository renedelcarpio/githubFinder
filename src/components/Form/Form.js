import { BiSearchAlt } from 'react-icons/bi';
import './Form.scss';

const Form = ({ onSubmit, value, onChange }) => {
  return (
    <>
      <h1 className="neon__title">Github Search</h1>
      <form onSubmit={onSubmit}>
        <input value={value} onChange={onChange} placeholder="Search by user" />
        <button type="submit">
          <BiSearchAlt className="search" />
        </button>
      </form>
    </>
  );
};

export default Form;
