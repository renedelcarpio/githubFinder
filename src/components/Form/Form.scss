@import '../../index.scss';

.neon__title {
  animation: flicker 1.5s infinite alternate;
  color: $color-light;
  font-size: 8rem;
  margin-top: 6rem;
}

@keyframes flicker {
  0%,
  18%,
  22%,
  25%,
  53%,
  57%,
  100% {
    text-shadow: 0 0 4px #fff, 0 0 11px #fff, 0 0 19px #fff,
      0 0 40px $color-cold1, 0 0 80px $color-cold1, 0 0 90px $color-cold1,
      0 0 100px $color-cold1, 0 0 150px $color-cold1;
  }
  20%,
  24%,
  55% {
    text-shadow: none;
  }
}

form {
  display: flex;
  flex-wrap: nowrap;
  margin: 10rem 0 5rem 0;
}

input {
  background: transparent;
  border: none;
  border-radius: 4rem;
  box-shadow: $box-shadow-dark-inset;
  color: $color-cold1;
  font-size: 1.6rem;
  margin-right: 1rem;
  padding: 1.2rem 1.5rem;
  outline: none;
}

button {
  display: flex;
  justify-content: center;
  align-items: center;
  background: transparent;
  border: none;
  border-radius: 50%;
  box-shadow: $box-shadow-dark;
  color: $color-dark1;
  height: 4.8rem;
  width: 4.8rem;

  &:active {
    box-shadow: $box-shadow-dark-inset;
    color: $color-cold1;
  }
}

.search {
  font-size: 2.4rem;
}

@media (max-width: 600px) {
  .neon__title {
    font-size: 3.8rem;
  }

  form {
    margin: 10rem 0 2rem 0;
  }

  input {
    font-size: 1.2rem;
  }

  button {
    height: 3.8rem;
    width: 3.8rem;
  }
}
