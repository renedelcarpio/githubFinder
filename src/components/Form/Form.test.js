import { shallow } from 'enzyme';
import Form from './Form';

const wrapper = shallow(<Form />);

describe('Test in <Form/> component', () => {
  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have the correct title', () => {
    const title = wrapper.find('.neon__title');
    expect(title.text()).toEqual('Github Search');
  });

  test('should have an imput', () => {
    const input = wrapper.find('input');
    expect(input.length).toBe(1);
  });

  test('should have a submit button', () => {
    const button = wrapper.find('button');
    expect(button.length).toBe(1);
  });
});
