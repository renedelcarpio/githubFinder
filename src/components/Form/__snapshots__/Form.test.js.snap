// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Form/> component should match the snapshot 1`] = `
<Fragment>
  <h1
    className="neon__title"
  >
    Github Search
  </h1>
  <form>
    <input
      placeholder="Search by user"
    />
    <button
      type="submit"
    >
      <BiSearchAlt
        className="search"
      />
    </button>
  </form>
</Fragment>
`;
